ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/ImageSingleVulnerabilities.tsx

```diff
@@ -1,13 +1,208 @@
+
+function severityCountsFromImageVulnerabilities(
+    imageVulnerabilities: ImageVulnerabilitiesResponse['image']['imageVulnerabilities']
+): Record<VulnerabilitySeverity, number> {
+    const severityCounts = {
+        LOW_VULNERABILITY_SEVERITY: 0,
+        MODERATE_VULNERABILITY_SEVERITY: 0,
+        IMPORTANT_VULNERABILITY_SEVERITY: 0,
+        CRITICAL_VULNERABILITY_SEVERITY: 0,
+    };
+
+    imageVulnerabilities.forEach(({ severity }) => {
+        severityCounts[severity] += 1;
+    });
+
+    return severityCounts;
+}
+
+function statusCountsFromImageVulnerabilities(
+    imageVulnerabilities: ImageVulnerabilitiesResponse['image']['imageVulnerabilities']
+): Record<FixableStatus, number> {
+    const statusCounts = {
+        Fixable: 0,
+        'Not fixable': 0,
+    };
+
+    imageVulnerabilities.forEach(({ isFixable }) => {
+        if (isFixable) {
+            statusCounts.Fixable += 1;
+        } else {
+            statusCounts['Not fixable'] += 1;
+        }
+    });
+
+    return statusCounts;
+}
+
+export type ImageVulnerabilitiesVariables = {
+    id: string;
+};
+
+export type ImageVulnerabilitiesResponse = {
+    image: {
+        imageVulnerabilities: {
+            severity: string;
+            isFixable: boolean;
+        }[];
+    };
+};
+
+export const imageVulnerabilitiesQuery = gql`
+    query getImageVulnerabilities($id: ID!) {
+        image(id: $id) {
+            id
+            imageVulnerabilities {
+                severity
+                isFixable
+            }
+        }
+    }
+`;
+
+export type ImageSingleVulnerabilitiesProps = {
+    imageId: string;
+};
+
+function ImageSingleVulnerabilities({ imageId }: ImageSingleVulnerabilitiesProps) {
+    // TODO Needs integration with URL search filter
+    const { data, loading, error } = useQuery<
+        ImageVulnerabilitiesResponse,
+        ImageVulnerabilitiesVariables
+    >(imageVulnerabilitiesQuery, {
+        variables: { id: imageId },
+    });
+
+    const [activeTabKey, setActiveTabKey] = useCveStatusTabParameter();
+
+    const handleTabClick: TabsProps['onSelect'] = (e, tabKey) => {
+        if (isValidCveStatusTab(tabKey)) {
+            setActiveTabKey(tabKey);
+        }
+    };
+
+    let mainContent: ReactNode | null = null;
+
+    if (error) {
+        mainContent = (
+            <Bullseye>
+                <EmptyState variant={EmptyStateVariant.large}>
+                    <EmptyStateIcon
+                        className="pf-u-danger-color-100"
+                        icon={ExclamationCircleIcon}
+                    />
+                    <Title headingLevel="h2">{getAxiosErrorMessage(error)}</Title>
+                    <EmptyStateBody>Adjust your filters and try again</EmptyStateBody>
+                </EmptyState>
+            </Bullseye>
+        );
+    } else if (loading && !data) {
+        mainContent = (
+            <Bullseye>
+                <Spinner isSVG />
+            </Bullseye>
```

Comment:
just curious, what's the difference in logic here between using a spinner vs the skeleton?

Reply:
I went with the spinner since we are waiting on a large chunk of UI with variable shape. (Summary cards + a table with an unknown number of rows.) That and the spinner is what we use in other parts of the app for these types of table sections, and it matches with the [PF example](https://www.patternfly.org/v4/components/spinner/design-guidelines#spinner-in-a-data-list-or-table-large).

I try to use the Skeleton for small, precise UI elements that are loading that we will definitely know the shape of, like the header/label section at the top of this page, or the result preview items on the collection form.